File: samples/node.js/hyperdocs.js
var request  = require("request");
var fs = require("fs");
var credentials = require("./credentials.json");

var appId = credentials.appId
var appKey = credentials.appKey

var validEndpoints = ["readPAN", "readPassport", "readAadhaar", "readKYC"];
function requestFn(filePath, fileType, headerParams, bodyParams, endPoint, cb){
    if(fileType != "image" && fileType != "pdf") return cb("unsupported file type");
    if(validEndpoints.indexOf(endPoint) == -1) return cb("invalid endPoint");
    try{
        fs.statSync(filePath);
    }catch(e){
        return cb("invalid file path");
    }
    var formData = bodyParams;
    formData[fileType] = fs.createReadStream(filePath);

    request({
        url: 'https://ind-docs.hyperverge.co/v2.0/'+endPoint,
        method: 'POST',
        formData: formData,
        headers: {
            'appid' : appId,
            'appkey' : appKey,
            'Accept' : "application/json",
            ...headerParams
        },
        json: true
    }, function(err, response, body){
        cb(err, body)
    });
}

module.exports = {
    readKYCImage : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "image", headerParams, bodyParams, "readKYC", cb) },
    readKYCpdf : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "pdf", headerParams, bodyParams, "readKYC", cb) },
    readPanImage : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "image", headerParams, bodyParams, "readPAN", cb) },
    readPanAadhaar : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "pdf", headerParams, bodyParams, "readPAN", cb) },
    readPassportAadhaar : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "image", headerParams, bodyParams, "readPassport", cb) },
    readPassportPdf : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "pdf", headerParams, bodyParams, "readPassport", cb) },
    readAadhaarImage : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "image", headerParams, bodyParams, "readAadhaar", cb) },
    readAadhaarPdf : function(filePath, headerParams, bodyParams, cb){ requestFn(filePath, "pdf", headerParams, bodyParams, "readAadhaar", cb) },
}

File: samples/python/hyperdocs.py
import requests
import os

import json

with open('config.json', 'r') as f:
 config = json.load(f)

url = "https://ind-docs.hyperverge.co/v2.0/"


appId = config['appId']
appKey = config['appKey']


#requestFn("./sample.jpg", "image", "readKYC")
#requestFn("./sample.pdf", "pdf", "readKYC")
# valid file types : image, pdf
# valid endPoints : readKYC, readPAN, readPassport, readAadhaar
# Though readKYC works on all documents. Use the appropriate endPoint if the document
# type is known as this would provide a marginally higher accuracy and better
# performance.
def requestFn(filePath, fileType, headerParams, bodyParams, endPoint):
 if(not os.path.exists(filePath)):
  return dict(error="Invalid file path")
 headers = {
     'appid':  appId,
     'appkey': appKey
     }
 if(fileType == "image"):
  response = requests.post(url+endPoint, files=dict(image=open(filePath)), data=bodyParams, headers=dict(headers, **headerParams));
 elif(fileType == "pdf"):
  response = requests.post(url+endPoint, files=dict(pdf=open(filePath)), data=bodyParams, headers=dict(headers, **headerParams));
 else:
  return dict(error='Invalid file type')
 return dict(response=response.text)

